lightning/src/events/mod.rs

```diff
@@ -1540,20 +1542,29 @@ impl Writeable for Event {
 				}
 				let skimmed_fee_opt = if counterparty_skimmed_fee_msat == 0 { None }
 					else { Some(counterparty_skimmed_fee_msat) };
+
+				let via_channel_id_legacy = via_channel_ids.last().cloned();
+				let via_user_channel_id_legacy = via_user_channel_ids.last().cloned();
 				write_tlv_fields!(writer, {
 					(0, payment_hash, required),
 					(1, receiver_node_id, option),
 					(2, payment_secret, option),
-					(3, via_channel_id, option),
+					// Marked as legacy in version 0.2.0; superseded by `via_channel_ids`, which
+					// includes all channel IDs used in the payment instead of only the last one.
+					(3, via_channel_id_legacy, option),
 					(4, amount_msat, required),
-					(5, via_user_channel_id, option),
+					// Marked as legacy in version 0.2.0 for the same reason as `via_channel_id_legacy`;
+					// superseded by `via_user_channel_ids`.
+					(5, via_user_channel_id_legacy, option),
 					// Type 6 was `user_payment_id` on 0.0.103 and earlier
 					(7, claim_deadline, option),
 					(8, payment_preimage, option),
 					(9, onion_fields, option),
 					(10, skimmed_fee_opt, option),
 					(11, payment_context, option),
 					(13, payment_id, option),
+					(15, *via_channel_ids, optional_vec),
+					(17, *via_user_channel_ids, optional_vec),
```

Comment:
When I comment out these two new lines, no tests seem to fail. Probably no coverage of this change?

Reply:
Thanks a lot for the pointer!
Updated the test-suite coverage, so that we check for proper event serialisation-deserialisation in [pr3655.08](https://github.com/shaavan/rust-lightning/commits/pr3655.08)